app/code/Magento/CatalogUrlRewrite/Observer/CategoryUrlPathAutogeneratorObserver.php

```diff
@@ -90,6 +100,12 @@ public function execute(\Magento\Framework\Event\Observer $observer)
                 $resultUrlKey = $category->formatUrlKey($category->getOrigData('name'));
                 $this->updateUrlKey($category, $resultUrlKey);
             }
+            if ($category->hasChildren()) {
+                $defaultUrlKey = $this->getDefaultUrlKey->execute((int)$category->getId());
+                if (!empty($defaultUrlKey)) {
```

Comment:
Could be simplified to:
```suggestion
                if ($defaultUrlKey) {
```

Reply:
Fixed